ingestion/src/metadata/ingestion/source/metadata/alationsink/metadata.py

```diff
@@ -0,0 +1,397 @@
+                return True
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Failed to check null type for {model_str(om_column.name)}: {exc}"
+            )
+        return None
+
+    def create_column_request(
+        self,
+        alation_datasource_id: int,
+        schema_name: str,
+        table_name: str,
+        om_column: Column,
+    ) -> Optional[CreateColumnRequest]:
+        """
+        Method to form the CreateColumnRequest object
+        """
+        try:
+            return CreateColumnRequest(
+                key=fqn._build(  # pylint: disable=protected-access
+                    str(alation_datasource_id),
+                    schema_name,
+                    table_name,
+                    model_str(om_column.name),
+                ),
+                column_type=om_column.dataType.value.lower(),
+                title=model_str(om_column.name),
+                description=model_str(om_column.description),
+                position=str(om_column.ordinalPosition)
+                if om_column.ordinalPosition
+                else None,
+                index=self._get_column_index(om_column),
+                nullable=self._check_nullable_column(om_column),
+            )
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Failed to create column request for {model_str(om_column.name)}: {exc}"
+            )
+        return None
+
+    def ingest_columns(
+        self, alation_datasource_id: int, schema_name: str, om_table: Table
+    ):
+        """
+        Method to ingest the columns
+        """
+        try:
+            create_requests = CreateColumnRequestList(root=[])
+            for om_column in om_table.columns or []:
+                create_column_request = self.create_column_request(
+                    alation_datasource_id=alation_datasource_id,
+                    schema_name=schema_name,
+                    table_name=model_str(om_table.name),
+                    om_column=om_column,
+                )
+                if create_column_request:
+                    create_requests.root.append(create_column_request)
+            if create_requests.root:
+                # Make the API call to write the columns to Alation
+                self.alation_sink_client.write_entities(
+                    alation_datasource_id, create_requests
+                )
+        except Exception as exc:
+            logger.debug(traceback.format_exc())
+            logger.error(
+                f"Unable to ingest columns for table [{model_str(om_table.name)}]: {exc}"
+            )
+
+    def ingest_tables(self, alation_datasource_id: int, om_schema: DatabaseSchema):
+        """
+        Method to ingest the tables
+        """
+        try:
+            # Iterate over all the tables in OpenMetadata
+            om_tables = list(
```

Comment:
do we need to force the list here?

Reply:
Yes the list gets used again later [here](https://github.com/open-metadata/OpenMetadata/pull/16982/files#diff-b8127404335acba3b5bb1df9128cdb3928a500d205dbab6026695c352837a506R291)